typescript/phoenix-sdk/src/market.ts

```diff
@@ -94,53 +91,51 @@ export class Market {
    * @param marketAddress The `PublicKey` of the market account
    */
   static async load({
-    connection,
     address,
+    buffer,
+    tokenList,
   }: {
-    connection: Connection;
     address: PublicKey;
+    buffer: Buffer;
+    tokenList: TokenConfig[];
   }): Promise<Market> {
-    // Fetch the account data for the market
-    const account = await connection.getAccountInfo(address);
-    if (!account)
-      throw new Error("Account not found for market: " + address.toBase58());
-    const buffer = Buffer.from(account.data);
     const marketData = deserializeMarketData(buffer);
 
-    const allTokens =
-      CONFIG[getClusterFromEndpoint(connection.rpcEndpoint)].tokens;
-
-    const baseTokenConfig = allTokens.find(
+    const baseTokenConfig = tokenList.find(
       (token) => token.mint === marketData.header.baseParams.mintKey.toBase58()
     );
-    const quoteTokenConfig = allTokens.find(
+    const quoteTokenConfig = tokenList.find(
       (token) => token.mint === marketData.header.quoteParams.mintKey.toBase58()
     );
 
-    if (baseTokenConfig === undefined) {
-      throw new Error(
-        `Base token ${marketData.header.baseParams.mintKey} not found in config`
-      );
-    }
-    if (quoteTokenConfig === undefined) {
-      throw new Error(
-        `Quote token ${marketData.header.quoteParams.mintKey} not found in config`
-      );
-    }
+    const baseKey = marketData.header.baseParams.mintKey.toBase58();
+    const baseKeyNameBackup = baseKey.slice(0, 8) + "..." + baseKey.slice(-8);
+    const quoteKey = marketData.header.baseParams.mintKey.toBase58();
+    const quoteKeyNameBackup =
+      quoteKey.slice(0, 8) + "..." + quoteKey.slice(-8);
 
     const baseToken = new Token({
-      name: baseTokenConfig.name,
-      symbol: baseTokenConfig.symbol,
-      logoUri: baseTokenConfig.logoUri,
+      name:
+        baseTokenConfig !== undefined
```

Comment:
if token config is undefined we should fail. are there cases where this is expected?

Reply:
I think it's totally fine to use this with a market without a defined config! Don't see a reason why the API should be that restrictive